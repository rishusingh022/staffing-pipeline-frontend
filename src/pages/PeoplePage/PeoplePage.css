.people-page {
  display: flex;
  justify-content: center;
  flex-direction: column;
  justify-content: center;
  width: 100%;
}
.container-in-people {
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: var(--color-whitesmoke);
  overflow: scroll;
  flex-grow: 1;
  min-height: 70vh;
  padding-top: 30px;
  padding-bottom: 20px;
}

.people-page-body {
  background-color: var(--color-whitesmoke);
  min-height: 100vh;
}

.people-page-filter-container {
  display: flex;
  width: 100%;
  justify-content: center;
  gap: 0.75rem;
  padding: 2.5rem 2.5rem;
}

.people-page-filter-search-box {
  width: 30%;
  display: flex;
  flex-direction: row;
  gap: 1rem;
}

.people-page-filter-input-box {
  width: 100%;
  display: flex;
  flex-direction: row;
  gap: 1rem;
}
.people-page-filter-drop-downs {
  display: flex;
  flex-direction: row;
  gap: 1rem;
  align-items: center;
}

.people-page-filter-drop-downs button {
  padding: 10px 15px;
}

@media screen and (max-width: 768px) {
  .people-page-filter-container {
    flex-direction: column;
    padding: 1rem 0.5rem 1rem 18.7%;
  }
}
