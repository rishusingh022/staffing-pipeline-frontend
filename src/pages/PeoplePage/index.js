import * as React from 'react';

const LoginPage = () => {
  return (
    <div>
      <h1>People Page</h1>
    </div>
  );
};

export default LoginPage;
