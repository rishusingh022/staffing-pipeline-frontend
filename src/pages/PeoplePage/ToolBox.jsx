import React from 'react';
import Dropdown from '../../components/Dropdown';
import { Search } from '../../components';
import props from 'prop-types';
import './PeoplePage.css';
import Button from '../../components/Button';

function ToolBox({
  handleSearchChange,
  handleTechnologyChange,
  handleRoleChange,
  technologyOptions,
  roleOptions,
  navigate,
}) {
  const handleSearch = searchValue => {
    handleSearchChange(searchValue);
  };
  const selectTechOption = option => {
    handleTechnologyChange(option);
  };
  const selectRoleOption = option => {
    handleRoleChange(option);
  };
  const handleAddNewMember = () => {
    navigate('/users/add');
  };
  return (
    <div className="people-page-filter-container">
      <div className="people-page-filter-search-box">
        <div className="people-page-filter-input-box">
          <Search placeHolderValue={'Search for people'} handleSearch={handleSearch} />
        </div>
      </div>
      <div className="people-page-filter-drop-downs">
        <Dropdown dropdownName={'Technology'} dropdownData={technologyOptions} selectOption={selectTechOption} />
        <Dropdown dropdownName={'Role'} dropdownData={roleOptions} selectOption={selectRoleOption} />
        <Button buttonText={'Add new member'} handleClick={handleAddNewMember} />
      </div>
      <div></div>
    </div>
  );
}

ToolBox.propTypes = {
  handleSearchChange: props.func.isRequired,
  handleTechnologyChange: props.func.isRequired,
  handleRoleChange: props.func.isRequired,
  technologyOptions: props.array.isRequired,
  roleOptions: props.array.isRequired,
  navigate: props.func.isRequired,
};

export default ToolBox;
