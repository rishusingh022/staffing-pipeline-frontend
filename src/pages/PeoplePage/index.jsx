import * as React from 'react';
import './PeoplePage.css';
import { Header } from '../../components';

const PeoplePage = () => {
  return (
    <div>
      <Header hasNav={true} />
      <h1>People Page</h1>
    </div>
  );
};

export default PeoplePage;
