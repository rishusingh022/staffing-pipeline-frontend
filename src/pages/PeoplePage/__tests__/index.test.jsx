import { render } from '@testing-library/react';
import * as React from 'react';
import PeoplePage from '..';

jest.mock('react-router-dom');
describe('PeoplePage', () => {
  it('should render without crashing', () => {
    const { asFragment } = render(<PeoplePage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
