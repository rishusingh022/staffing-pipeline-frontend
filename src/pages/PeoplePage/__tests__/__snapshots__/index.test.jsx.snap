// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeoplePage should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="people-page"
  >
    <div
      class="font-light h-[15vh] text-white flex justify-center bg-deepBlue header-container"
    >
      <div
        class="h-full xl:w-[63vw] lg:[70vw] md:[80vw] flex box-border"
      >
        <img
          class="h-[70px] w-[155px] aspect-auto self-end pl-0 lg:mr-[180px] sm:mr-[90px] pt-0 pb-[5px]"
          src="McK_Logo.png"
        />
        <div
          class="grid grid-cols-3 gap-10 items-end"
        >
          <button
            class="text-cyan border-b-[5px] border-b-cyan border-solid pt-0 pb-2.5 px-[5px]"
          >
            Engagements
          </button>
          <button
            class="text-[white] hover:text-cyan hover:border-b-[5px] hover:border-b-cyan hover:border-solid pt-0 pb-[15px] hover:pb-2.5 px-[5px]"
          >
            People
          </button>
          <button
            class="text-[white] hover:text-cyan hover:border-b-[5px] hover:border-b-cyan hover:border-solid pt-0 pb-[15px] hover:pb-2.5 px-[5px]"
          >
            Case Studies
          </button>
        </div>
      </div>
    </div>
    <div
      class="people-page-filter-container"
    >
      <div
        class="people-page-filter-search-box"
      >
        <div
          class="people-page-filter-input-box"
        >
          <div
            class="search-container"
          >
            <input
              placeholder="Search for people ..."
              type="text"
            />
            <div
              class="filter-search-button"
            >
              <button
                class="button"
                data-testid="button"
              >
                Search
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="people-page-filter-drop-downs"
      >
        <div
          class="w-44 font-normal dropdown-container"
        >
          <div
            class="bg-white w-full p-2 dropdown-header
        flex justify-between items-center border-solid border border-black cursor-pointer"
          >
            Technology
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
              />
            </svg>
          </div>
          <div
            class="hidden"
          >
            <ul
              class="bg-white mt-2 overflow-y-auto max-h-60 border border-black-300"
            >
              <div
                class="flex items-center sticky top-0 h-10 bg-white"
              >
                <svg
                  class="w-10"
                  fill="currentColor"
                  height="18"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
                <input
                  class="placeholder:text-gray-700 outline-none text-mg w-36 pr-2"
                  placeholder="search"
                  type="text"
                  value=""
                />
              </div>
              <li
                class="p-2 text-sm hover:bg-gray-300 cursor-pointer
            block"
              >
                Node
              </li>
              <li
                class="p-2 text-sm hover:bg-gray-300 cursor-pointer
            block"
              >
                React Js
              </li>
              <li
                class="p-2 text-sm hover:bg-gray-300 cursor-pointer
            block"
              >
                Angular
              </li>
              <li
                class="p-2 text-sm hover:bg-gray-300 cursor-pointer
            block"
              >
                Java
              </li>
            </ul>
          </div>
        </div>
        <div
          class="w-44 font-normal dropdown-container"
        >
          <div
            class="bg-white w-full p-2 dropdown-header
        flex justify-between items-center border-solid border border-black cursor-pointer"
          >
            Role
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
              />
            </svg>
          </div>
          <div
            class="hidden"
          >
            <ul
              class="bg-white mt-2 overflow-y-auto max-h-60 border border-black-300"
            >
              <div
                class="flex items-center sticky top-0 h-10 bg-white"
              >
                <svg
                  class="w-10"
                  fill="currentColor"
                  height="18"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
                <input
                  class="placeholder:text-gray-700 outline-none text-mg w-36 pr-2"
                  placeholder="search"
                  type="text"
                  value=""
                />
              </div>
              <li
                class="p-2 text-sm hover:bg-gray-300 cursor-pointer
            block"
              >
                Junior Engineer
              </li>
              <li
                class="p-2 text-sm hover:bg-gray-300 cursor-pointer
            block"
              >
                Engineer1
              </li>
              <li
                class="p-2 text-sm hover:bg-gray-300 cursor-pointer
            block"
              >
                Engineer2
              </li>
            </ul>
          </div>
        </div>
        <button
          class="button"
          data-testid="button"
        >
          Add new member
        </button>
      </div>
      <div />
    </div>
    <div
      class="container-in-people"
    >
      <div
        class="grid grid-cols-1 h-full gap-x-14 gap-y-8 lg:grid-cols-4 md:grid-cols-2"
      />
    </div>
  </div>
</DocumentFragment>
`;
